index.js

```diff
@@ -2787,6 +2787,28 @@
   }
   S.prop = def('prop', {a: [Accessible]}, [$.String, a, b], prop);
 
+  //# props :: Accessible a => Array String -> a -> b
+  //.
+  //. Takes an array of property names and an object with known properties and
+  //. returns the value of the specified property. If for some reason the
+  //. object lacks the specified property, a type error is thrown.
+  //.
+  //. For accessing properties of uncertain objects, use [`gets`](#gets)
+  //. instead.
+  //.
+  //. ```javascript
+  //. > S.props(['a', 'b', 'c'], {a: {b: {c: 1}}})
+  //. 1
+  //. ```
+  function props(path, obj) {
+    return path.reduce(function(memo, key) {
+      if (key in memo) return memo[key];
+      throw new TypeError('‘props’ expected object to have a property at ‘' +
+        path.join('.') + '’; ' + R.toString(obj) + ' does not');
```

Comment:
I don't think we should use `path.join('.')`. It means `['foo.bar', 'baz', 'quux']` is represented as `foo.bar.baz.quux`, which is misleading. Furthermore, dot notation is not valid for all property names. I think `["a", "b", "c"]` is the most appropriate format.